constraints/src/main/java/gov/nasa/jpl/aerie/constraints/model/EvaluationEnvironment.java

```diff
@@ -8,10 +9,15 @@
 public record EvaluationEnvironment(
     Map<String, ActivityInstance> activityInstances,
     Map<String, Spans> spansInstances,
+    Map<String, Interval> intervals,
```

Comment:
Can you explain why you needed to add this ?

Reply:
I got tired of using collections to store single objects. Like using a Windows to store something that is guaranteed to be a single Window, or a Spans to store a guaranteed single Span. So I decided the most natural argument for the CoexGoal template factory would be an `Interval` instead of a `Windows` that only has one window. This caused all the Interval-related changes to the edsl. It also meant we need to store intervals in the environment so CoexGoal can generate the Interval, and the ast nodes in the factory can access it.